src/azul/__init__.py

```diff
@@ -155,13 +159,13 @@ def dss_checkout_bucket(self, dss_endpoint: Optional[str] = None):
             dss_endpoint = self.dss_endpoint
         return self._dss_bucket(dss_endpoint, qualifier='checkout')
 
+    @lru_cache(maxsize=10)  # Maxsize should be grater than the number of permutations for this method
     def _dss_bucket(self, dss_endpoint: str, qualifier=None):
+        ssm = boto3.client('ssm')
         stage = self._dss_deployment_stage(dss_endpoint)
-        # For domain_part, DSS went from `humancellatlas` to `hca` in 9/2018 and started reverting back to
-        # `humancellatlas` in 12/2018. As I write this, only `dev` is back on `humancellatlas`
-        domain_part = 'hca' if stage == 'prod' else 'humancellatlas'
-        qualifier = [qualifier] if qualifier else []
-        return '-'.join(['org', domain_part, 'dss', *qualifier, stage])
+        name = f'dcp/dss/{stage}/environment'
+        dss_config = json.loads(ssm.get_parameter(Name=name)['Parameter']['Value'])
+        return dss_config['DSS_S3_BUCKET'] if not qualifier else dss_config['DSS_S3_CHECKOUT_BUCKET']
```

Comment:
This is very brittle. What if we add a third value for qualifier? Which values for the qualifier argument do you currently support: None and 'checkout'. Any other value should raise an exception. 

Alternatively, you could change the type of that argument so it really only supports the two cases, normal bucket or checkout bucket.

BUT: Note how the qualifier value actually occurs in the name of the variable. Just in uppercase. Study the old logic and adjust it.